### DISCUSSION TITLE
How to know when clip is available at url

### DISCUSSION THREAD

**Discussion by TheQue42** [blakeblackshear/frigate#1151] [General]
First off of course, thanks for the amazing application :) 

I've just started using frigate, and have automations in HA triggering on the mqtt messages, and now I have been fiddling around how not to get swamped with tons of videos to my telegram account, when there is prolonged movement in front of the camera.
I do have a working hack, but one annoying issue left. 

I cant clearly figure out when the moveclip is actually available to fetch? I'm filtering on frigate/events, with payload=end since my understanding that indicates the end of the object movement event in some way, and I get the URL via

`api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4`

but trying to send that to telegram_bot, will normally fail, with telegram trying to fetch from the url 5 times really quickly, and then giving up.

Instead I have an annoying "wait-16 seconds", and THEN send.

Is there a way to know for sure, when the clip is available?

**Comment by blakeblackshear** [comment]
Not yet. You have to poll the endpoints until it doesn't return a 404. Next release will include a 'saved' event type.

**Reply by TheQue42** [reply]
I have entered a feature request for the telegram_bot, to make it configurable how many attempts it does, or at least the delay between them (if i remember correctly), but,...

Anyway, the fact that such a feature is on its way, is superb! Thanks!

**Comment by CMDR-Sloma** [comment]
Sorry for resurrecting this topic, but has this been implemented yet?

**Reply by NickM-27** [reply]
A saved endpoint hasn't been implemented but the event recordings are available dynamically. I have no wait for mine and it works as expected.

**Reply by TheQue42** [reply]
Personally I have just detected that the clips I send on telegram are sometimes shorter than the "real" clip. I forward the link `https://example.se/api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4`, to both telegrams URL parameter, and as pure text in the caption of the message.
Sometimes, there is a difference of tens of seconds between the clip that is sent to telegram, and watching it via the link in a browser.

So I would still _very_ much like the "saved" flag.

**Reply by TheQue42** [reply]
Addionally, there doesnt seem to be an actual feature request for this? Since @blakeblackshear  mentioned earlier "next release", I thought there might be, but my quick search doesnt seem to be one?

**Reply by NickM-27** [reply]
> Sometimes, there is a difference of tens of seconds between the clip that is sent to telegram, and watching it via the link in a browser.

This would be because the last 10 second segment hasn't made it out of the cache. 

And https://github.com/blakeblackshear/frigate/issues/2128 covers this request
